/*
 * Copyright 2024 Sowers, LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
syntax = "proto3";

option go_package = "bosca.io/api/protobuf/bosca/content";

package bosca.content;

message Traits {
  repeated Trait traits = 1;
}

message Trait {
  string id = 1;
  string name = 2;
  string description = 3;
  repeated string workflow_ids = 4;
}

message TraitWorkflowIdRequest {
  string trait_id = 1;
  string workflow_id = 2;
}

message TraitWorkflowActivityIdRequest {
  string trait_id = 1;
  string workflow_id = 2;
  string activity_id = 3;
}
